datafusion/core/src/physical_plan/mod.rs

```diff
@@ -477,6 +473,66 @@ impl Partitioning {
             RoundRobinBatch(n) | Hash(_, n) | UnknownPartitioning(n) => *n,
         }
     }
+
+    /// Returns true when the guarantees made by this [[Partitioning]] are sufficient to
+    /// satisfy the partitioning scheme mandated by the `required` [[Distribution]]
+    pub fn satisfy<F: FnOnce() -> EquivalenceProperties>(
+        &self,
+        required: Distribution,
+        equal_properties: F,
+    ) -> bool {
+        match required {
+            Distribution::UnspecifiedDistribution => true,
```

Comment:
we can add some UT for `satisfy()` to confirm some cases:
- UnspecifiedDistribution - UnspecifiedDistribution
- Hash - SinglePartition
- Hash - Hash
- .......

Reply:
SinglePartition should already be covered by existing UTs.

Reply:
Perhaps @jackwener  was getting at adding additional testing could also help to document the intent of satisfy() as well as making the coverage was clear, even if they did not add additional coverage

I would be interested in helping to write such tests as part of my review of this code (as it would help me gain a better understanding)